src/main/scala/is/hail/variant/AltAlleleView.scala

```diff
@@ -0,0 +1,99 @@
+package is.hail.variant
+
+import is.hail.annotations._
+import is.hail.expr._
+import is.hail.utils._
+
+class AltAlleleView(taa: TAltAllele) extends View {
+  private val t = taa.fundamentalType.asInstanceOf[TStruct]
+  private val refIdx = t.fieldIdx("ref")
+  private val altIdx = t.fieldIdx("alt")
+  private var region: MemoryBuffer = _
+  private var offset: Long = _
+  private var _ref: String = null
+  private var _alt: String = null
+
+  assert(t.isFieldRequired(refIdx))
+  assert(t.isFieldRequired(altIdx))
+
+  def setRegion(region: MemoryBuffer, offset: Long) {
+    this.region = region
+    this.offset = offset
+    this._ref = null
+    this._alt = null
+  }
+
+  def getRef(): String = {
+    if (_ref == null)
+      _ref = TString.loadString(region, t.loadField(region, offset, refIdx))
+    _ref
+  }
+
+  def getAlt(): String = {
+    if (_alt == null)
+      _alt = TString.loadString(region, t.loadField(region, offset, altIdx))
+    _alt
+  }
+
+  import AltAlleleType._
+
+  def altAlleleType: AltAlleleType = {
```

Comment:
Hmm.  This duplicates a ton of code.  What about making `AltAllele` abstract and having a `ConcreteAltAllele` and a `RegionValueAltAllele`, the latter of which is essentially what you've written?